decoders/netflow/netflow.go

```diff
@@ -355,15 +355,26 @@ func (ts *BasicTemplateSystem) AddTemplate(version uint16, obsDomainId uint32, t
 		ts.templates[version][obsDomainId] = make(map[uint16]interface{})
 	}
 	var templateId uint16
-	switch templateIdConv := template.(type) {
+	switch record := template.(type) {
 	case IPFIXOptionsTemplateRecord:
```

Comment:
You're duplicating code in both cases. Merge them into one block:
```go
case IPFIXOptionsTemplateRecord, NFv9OptionsTemplateRecord:
```

Reply:
I tried, but the compiler does not allow to do this:
![Screenshot from 2024-09-27 16-21-12](https://github.com/user-attachments/assets/bcba16c2-cde8-45ee-a40a-77e2cd323b9b)

Reply:
Surprisingly, it doesn't work with `type switch`. Sorry for having bothered you.

Reply:
Surprisingly, it didn’t work with a `type switch`. Sorry for having bothered you.